Library/Homebrew/cleanup.rb

```diff
@@ -245,21 +245,16 @@ def cleanup_path(path)
     end
 
     def cleanup_lockfiles(*lockfiles)
-      return unless HOMEBREW_LOCK_DIR.directory?
+      return if dry_run?
 
-      if lockfiles.empty?
+      if lockfiles.empty? && HOMEBREW_LOCK_DIR.directory?
         lockfiles = HOMEBREW_LOCK_DIR.children.select(&:file?)
       end
 
       lockfiles.each do |file|
         next unless file.readable?
         next unless file.open(File::RDWR).flock(File::LOCK_EX | File::LOCK_NB)
-
-        begin
-          cleanup_path(file) { file.unlink }
-        ensure
-          file.open(File::RDWR).flock(File::LOCK_UN) if file.exist?
```

Comment:
How about keeping this line and the `ensure`?

I feel it makes the code easier to understand: it’s clear where the `LOCK_UN` happens, and it visually complements the `LOCK_EX`/`LOCK_NB`. Also, graceful degradation (if that’s desirable here).

Reply:
Readded the unlock.